presto-main/src/main/java/io/prestosql/memory/QueryContext.java

```diff
@@ -70,6 +70,10 @@
     private final SpillSpaceTracker spillSpaceTracker;
     private final Map<TaskId, TaskContext> taskContexts = new ConcurrentHashMap<>();
 
+    @GuardedBy("this")
+    private boolean isResourceOverCommit;
+    private boolean memoryLimitsInitialized;
```

Comment:
this should be `volatile`

Reply:
Made volatile (even though it's safe enough to do a redundant initialization and maybe slight cheaper as a non-volatile)